commit 37a49f3ae3bdfb2942b7145457fb456745b18bc5
Author: Po-Hsien Wang <pwang@chromium.org>
Date:   Wed Nov 1 17:19:26 2017 -0700

    arc-mesa: Disable support for GLES > 2.0
    
    Disable support for GLES > 2.0
    
    Change-Id: I428a56e08b473a106093b71e01094f3db6875cfb

diff --git a/src/egl/main/eglcontext.c b/src/egl/main/eglcontext.c
index 1a8e9bda1a..d134f72348 100644
--- a/src/egl/main/eglcontext.c
+++ b/src/egl/main/eglcontext.c
@@ -467,12 +467,7 @@ _eglParseContextAttribList(_EGLContext *ctx, _EGLDisplay *dpy,
             err = EGL_BAD_MATCH;
          break;
 
-      case 3:
-         /* Don't put additional version checks here.  We don't know that
-          * there won't be versions > 3.0.
-          */
-         break;
-
+         // Hack: Disallow creation contexts newer than 2.0.
       default:
          err = EGL_BAD_MATCH;
          break;
diff --git a/src/mesa/main/version.c b/src/mesa/main/version.c
index 34f8bbbcb1..f7d95b2d45 100644
--- a/src/mesa/main/version.c
+++ b/src/mesa/main/version.c
@@ -593,6 +593,9 @@ _mesa_compute_version(struct gl_context *ctx)
       return;
 
    ctx->Version = _mesa_get_version(&ctx->Extensions, &ctx->Const, ctx->API);
+   // HACK: Downgrade GLES2 version to 2.0.
+   if (ctx->API == API_OPENGLES2 && ctx->Version > 20)
+       ctx->Version = 20;
    ctx->Extensions.Version = ctx->Version;
 
    /* Make sure that the GLSL version lines up with the GL version. In some
